students/BrianNisonger/Lesson4/basic_operations.py

```diff
@@ -0,0 +1,52 @@
+from peewee import *
+from customer_model import Customer
+from customer_model import database
+
+
+def add_customer(**kwargs):
+    try:
+        kwargs['credit_limit'] = float(kwargs['credit_limit'])
+    except ValueError as err:
+        raise
+    if not database.table_exists('Customer'):
+        database.create_tables([Customer])
+    new_customer = Customer.insert(kwargs)
+    new_customer.execute()
+
+
+def search_customer(customer_id=None):
+    search_customer = Customer.select().where(
+        Customer.customer_id == customer_id).dicts()
+    field_names = Customer._meta.sorted_field_names
+    if search_customer:
+        return {
+            field_name: record[field_name]
+            for record in search_customer for field_name in field_names
+        }
+    else:
+        return {}
+
+
+def delete_customer(customer_id=None):
+    delete_customer = Customer.get_or_none(Customer.customer_id == customer_id)
+    if delete_customer:
+        delete_customer.delete_instance()
+
+
+def update_customer(**kwargs):
+    update_customer = Customer.get_or_none(
+        Customer.customer_id == kwargs['customer_id'])
+    if update_customer:
+        insert_dict = {
+            kwarg: kwargs[kwarg]
+            for kwarg in kwargs if kwarg != 'customer_id'
+        }
```

Comment:
This is a little hard to read, the kwarg kwargs kind of run together, consider
```
insert_dict = {key: kwargs[key] for key in kwargs if key != 'customer_id'}
```
or 
```
insert_dict = {key:value for key, value in kwargs.items() if key != "customer_id}
```
But, nice use of a dictionary comprehension. :)